Plurals fails to compile with duplicate `__var_count` generic

Been struggling to get this to compile in my project. I'm not sure if I'm just using this feature wrong, or if there's a bug in the implementation.

It seems to be caused when there is only one locale available.

Here's a minimal reproduction:

`Cargo.toml`
```toml
[dependencies]
leptos = { version = "0.5.0", features = ["csr"] }
leptos_meta = { version = "0.5.0", features = ["csr"] }
leptos_i18n = { path = "../../leptos_i18n", default-features = false, features = [
    "debug_interpolations",
    "csr",
    "cookie",
    "yaml_files",
] }
serde = { version = "1", features = ["derive"] }
console_error_panic_hook = { version = "0.1" }
wasm-bindgen = { version = "=0.2.87" }

[package.metadata.leptos-i18n]
default = "en"
locales = ["en"]
```

`src/app.rs`
```rust
#[component]
pub fn App() -> impl IntoView {
    leptos_meta::provide_meta_context();

    let i18n = provide_i18n_context();

    view! {
        <p>{t!(i18n, result_summary, count = || 5, query_str = || "test")}</p>
    }
}
```

`locales/en.yaml`
```yaml
---
result_summary:
  - "u64"
  - ["No results for {{ query_str }}:", 0]
  - ["1 result for {{ query_str }}:", 1]
  - ["{{ count }} results for {{ query_str }}:"]
```

```txt
❯ trunk build
2023-10-27T22:24:07.047974Z  INFO 📦 starting build
2023-10-27T22:24:07.048390Z  INFO spawning asset pipelines
2023-10-27T22:24:07.160652Z  INFO building yaml
   Compiling yaml v0.1.0 (/Users/lpetherbridge/pdev/leptos_i18n/examples/yaml)
error[E0403]: the name `__var_count` is already used for a generic parameter in this item's generic parameters
 --> src/main.rs:4:1
  |
4 | leptos_i18n::load_locales!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | already used
  | first use of `__var_count`
  |
  = note: this error originates in the macro `leptos_i18n::load_locales` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0416]: identifier `var_count` is bound more than once in the same pattern
 --> src/main.rs:4:1
  |
4 | leptos_i18n::load_locales!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ used in a pattern more than once

error[E0124]: field `var_count` is already declared
 --> src/main.rs:4:1
  |
4 | leptos_i18n::load_locales!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | field already declared
  | `var_count` first declared here
  |
  = note: this error originates in the macro `leptos_i18n::load_locales` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0124, E0403, E0416.
For more information about an error, try `rustc --explain E0124`.
error: could not compile `yaml` (bin "yaml") due to 3 previous errors
```

--- a/leptos_i18n_macro/src/load_locales/error.rs
+++ b/leptos_i18n_macro/src/load_locales/error.rs
@@ -45,7 +45,6 @@ pub enum Error {
     },
     ImpossibleRange(String),
     PluralTypeMissmatch {
-        locale: Rc<Key>,
         key_path: KeyPath,
         type1: PluralType,
         type2: PluralType,
@@ -143,8 +142,8 @@ impl Display for Error {
                 "Found duplicates namespaces in configuration (Cargo.toml): {:?}", 
                 duplicates
             ),
-            Error::PluralTypeMissmatch { locale, key_path, type1, type2 } => write!(f, "Missmatch plural value type as key {}, locale {:?} has type {} but another locale has type {}", key_path, locale, type1, type2),
-            Error::InvalidKey(key) => write!(f, "invalid key {:?}, it can't be used as a rust identifier, try removing whitespaces and special characters", key),
+            Error::PluralTypeMissmatch { key_path, type1, type2 } => write!(f, "Conflicting plural value type at key {}, found type {} but also type {}.", key_path, type1, type2),
+            Error::InvalidKey(key) => write!(f, "invalid key {:?}, it can't be used as a rust identifier, try removing whitespaces and special characters.", key),
             Error::EmptyPlural => write!(f, "empty plurals are not allowed"),
             Error::InvalidPluralType(t) => write!(f, "invalid plural type {:?}", t),
             Error::NestedPlurals => write!(f, "nested plurals are not allowed"),

--- a/leptos_i18n_macro/src/load_locales/locale.rs
+++ b/leptos_i18n_macro/src/load_locales/locale.rs
@@ -32,7 +32,7 @@ pub enum LocalesOrNamespaces {
     Locales(Vec<Locale>),
 }
 
-#[derive(Default)]
+#[derive(Default, Debug)]
 pub struct BuildersKeysInner(pub HashMap<Rc<Key>, LocaleValue>);
 
 pub enum BuildersKeys<'a> {
@@ -269,6 +269,8 @@ impl Locale {
             )?;
         }
 
+        default_keys.check_conflicts(&mut key_path)?;
+
         Ok(default_keys)
     }
 
@@ -293,12 +295,63 @@ impl Locale {
     }
 }
 
+impl BuildersKeysInner {
+    fn check_conflicts(&mut self, key_path: &mut KeyPath) -> Result<()> {
+        for (key, values) in &mut self.0 {
+            key_path.push_key(Rc::clone(key));
+            values.check_conflicts(key_path)?;
+            key_path.pop_key();
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
 pub enum LocaleValue {
     Value(Option<HashSet<InterpolateKey>>),
     Subkeys {
         locales: Vec<Locale>,
         keys: BuildersKeysInner,
     },
+}
+
+impl LocaleValue {
+    fn check_conflicts(&mut self, key_path: &mut KeyPath) -> Result<()> {
+        match self {
+            LocaleValue::Value(Some(keys)) => {
+                let mut iter = keys.iter();
+                let Some(count_type) = iter.find_map(|key| match key {
+                    InterpolateKey::Count(plural_type) => Some(*plural_type),
+                    _ => None,
+                }) else {
+                    return Ok(());
+                };
+
+                let other_type = iter.find_map(|key| match key {
+                    InterpolateKey::Count(plural_type) => Some(*plural_type),
+                    _ => None,
+                });
+
+                if let Some(other_type) = other_type {
+                    return Err(Error::PluralTypeMissmatch {
+                        key_path: std::mem::take(key_path),
+                        type1: count_type,
+                        type2: other_type,
+                    });
+                }
+
+                // if the set contains InterpolateKey::Count, remove variable keys with name "count"
+                // ("var_count" with the rename)
+                keys.retain(
+                    |key| !matches!(key, InterpolateKey::Variable(key) if key.name == "var_count"),
+                );
+
+                Ok(())
+            }
+            LocaleValue::Value(None) => Ok(()),
+            LocaleValue::Subkeys { keys, .. } => keys.check_conflicts(key_path),
+        }
+    }
 }
 
 #[derive(Debug, Clone)]

--- a/leptos_i18n_macro/src/load_locales/parsed_value.rs
+++ b/leptos_i18n_macro/src/load_locales/parsed_value.rs
@@ -267,45 +267,6 @@ impl ParsedValue {
         }
     }
 
-    fn merge_inner(
-        &self,
-        keys: &mut Option<HashSet<InterpolateKey>>,
-        top_locale: Rc<Key>,
-        key_path: &mut KeyPath,
-    ) -> Result<()> {
-        self.get_keys_inner(keys);
-        let Some(keys) = keys else {
-            return Ok(());
-        };
-        let mut iter = keys.iter();
-        let Some(count_type) = iter.find_map(|key| match key {
-            InterpolateKey::Count(plural_type) => Some(*plural_type),
-            _ => None,
-        }) else {
-            return Ok(());
-        };
-
-        let other_type = iter.find_map(|key| match key {
-            InterpolateKey::Count(plural_type) if *plural_type != count_type => Some(*plural_type),
-            _ => None,
-        });
-
-        if let Some(other_type) = other_type {
-            return Err(Error::PluralTypeMissmatch {
-                locale: top_locale,
-                key_path: std::mem::take(key_path),
-                type1: count_type,
-                type2: other_type,
-            });
-        }
-
-        // if the set contains InterpolateKey::Count, remove variable keys with name "count"
-        // ("var_count" with the rename)
-        keys.retain(|key| !matches!(key, InterpolateKey::Variable(key) if key.name == "var_count"));
-
-        Ok(())
-    }
-
     pub fn merge(
         &mut self,
         keys: &mut LocaleValue,
@@ -335,7 +296,10 @@ impl ParsedValue {
                 | ParsedValue::Variable(_)
                 | ParsedValue::ForeignKey(_),
                 LocaleValue::Value(keys),
-            ) => self.merge_inner(keys, top_locale, key_path),
+            ) => {
+                self.get_keys_inner(keys);
+                Ok(())
+            }
             // not compatible
             _ => Err(Error::SubKeyMissmatch {
                 locale: top_locale,

